packages/fxa-auth-server/test/local/routes/subscriptions/apple.js

```diff
@@ -0,0 +1,177 @@
+    Container.reset();
+    sinon.restore();
+  });
+
+  describe('registerOriginalTransactionId', () => {
+    const request = {
+      ...VALID_REQUEST,
+      params: { appName: 'test' },
+      payload: { originalTransactionId: 'testTransactionId' },
+    };
+
+    it('returns valid with new products', async () => {
+      appleIap.purchaseManager = {
+        registerToUserAccount: sinon.fake.resolves({}),
+      };
+      iapConfig.getBundleId = sinon.fake.resolves('testPackage');
+      const result = await appleIapHandler.registerOriginalTransactionId(
+        request
+      );
+      assert.calledOnce(appleIap.purchaseManager.registerToUserAccount);
+      assert.calledOnce(iapConfig.getBundleId);
+      assert.calledOnce(mockCapabilityService.iapUpdate);
+      assert.deepEqual(result, { transactionIdValid: true });
+    });
+
+    it('returns valid with no changed products', async () => {
+      mockCapabilityService.subscribedProductIds = sinon.fake.resolves([
+        'prod_1234',
+      ]);
+      appleIap.purchaseManager = {
+        registerToUserAccount: sinon.fake.resolves({}),
+      };
+      iapConfig.getBundleId = sinon.fake.resolves('testPackage');
+      const result = await appleIapHandler.registerOriginalTransactionId(
+        request
+      );
+      assert.calledOnce(appleIap.purchaseManager.registerToUserAccount);
+      assert.calledOnce(iapConfig.getBundleId);
+      assert.calledOnce(mockCapabilityService.iapUpdate);
+      assert.deepEqual(result, { transactionIdValid: true });
+    });
+
+    it('throws on invalid package', async () => {
+      appleIap.purchaseManager = {
+        registerToUserAccount: sinon.fake.resolves({}),
+      };
+      iapConfig.getBundleId = sinon.fake.resolves(undefined);
+      try {
+        await appleIapHandler.registerOriginalTransactionId(request);
+        assert.fail('Expected failure');
+      } catch (err) {
+        assert.calledOnce(iapConfig.getBundleId);
+        assert.strictEqual(err.errno, error.ERRNO.IAP_UNKNOWN_APPNAME);
+      }
+    });
+
+    it('throws on invalid transaction id', async () => {
+      const libraryError = new Error('Purchase is not registerable');
+      libraryError.name = PurchaseUpdateError.INVALID_ORIGINAL_TRANSACTION_ID;
+
+      appleIap.purchaseManager = {
+        registerToUserAccount: sinon.fake.rejects(libraryError),
+      };
+      iapConfig.getBundleId = sinon.fake.resolves('testPackage');
+      try {
+        await appleIapHandler.registerOriginalTransactionId(request);
+        assert.fail('Expected failure');
+      } catch (err) {
+        assert.strictEqual(err.errno, error.ERRNO.IAP_INVALID_TOKEN);
+        assert.calledOnce(appleIap.purchaseManager.registerToUserAccount);
+        assert.calledOnce(iapConfig.getBundleId);
+      }
+    });
+
+    it('throws on conflict', async () => {
+      const libraryError = new Error('Purchase is not registerable');
+      libraryError.name = PurchaseUpdateError.CONFLICT;
```

Comment:
Can you remind me when we might see a CONFLICT update error? Is this when the document is currently being accessed/updated?

Reply:
When the purchase was already registered to a different user.